staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -140,48 +138,55 @@ func GetResource(r rest.Getter, e rest.Exporter, scope RequestScope) restful.Rou
 }
 
 // GetResourceWithOptions returns a function that handles retrieving a single resource from a rest.Storage object.
-func GetResourceWithOptions(r rest.GetterWithOptions, scope RequestScope) restful.RouteFunction {
+func GetResourceWithOptions(r rest.GetterWithOptions, scope RequestScope, isSubresource bool) http.HandlerFunc {
 	return getResourceHandler(scope,
-		func(ctx request.Context, name string, req *restful.Request) (runtime.Object, error) {
+		func(ctx request.Context, name string, req *http.Request) (runtime.Object, error) {
 			opts, subpath, subpathKey := r.NewGetOptions()
-			if err := getRequestOptions(req, scope, opts, subpath, subpathKey); err != nil {
+			if err := getRequestOptions(req, scope, opts, subpath, subpathKey, isSubresource); err != nil {
 				return nil, err
 			}
 			return r.Get(ctx, name, opts)
 		})
 }
 
-func getRequestOptions(req *restful.Request, scope RequestScope, into runtime.Object, subpath bool, subpathKey string) error {
+// getRequestOptions parses out options and can include path information.  The path information shouldn't include the subresource.
+func getRequestOptions(req *http.Request, scope RequestScope, into runtime.Object, subpath bool, subpathKey string, isSubresource bool) error {
 	if into == nil {
 		return nil
 	}
 
-	query := req.Request.URL.Query()
+	query := req.URL.Query()
 	if subpath {
 		newQuery := make(url.Values)
 		for k, v := range query {
 			newQuery[k] = v
 		}
-		newQuery[subpathKey] = []string{req.PathParameter("path")}
+
+		ctx := scope.ContextFunc(req)
+		requestInfo, _ := request.RequestInfoFrom(ctx)
+		startingIndex := 2
+		if isSubresource {
```

Comment:
This looks tricky, but `path` was the "extra bits" from the go-restful request.  This does the same thing.